ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,138 @@
+    try {
+        //Delete thread when thread exists
+        if (timer_queue[secretName + '_' + appid] != undefined ) {
+            clearTimeout(timer_queue[secretName + '_' + appid].timer)
+        }
+        delete timer_queue[secretName + '_' + appid];
+        // An insurance mechanism to ensure that thread init_interval is active
+        init_secret_delete_thread(DB)
+    } catch (error) {
+        console.error('error removeTimer :: ', error)
+    }
+}
+
+/**
+ * Expose the interface externally, scheduled tasks of delete secret
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @param {Object} plannedDeleteTime : planned delete time for secret eg. '1659931644416'
+ * @returns
+ */
+const startDeleteSecretTask = (DB, appid, secretName, plannedDeleteTime) => {
+    try {
+        let delayMillisecond = plannedDeleteTime - new Date().getTime()
+        let needAdd = true
+        let key = secretName + '_' + appid
+        // process exist timer, 
+        // If the new time is within plus or minus 10 seconds of the original time, keep the original plan unchanged; otherwise, clear the original plan and add again
+        if (timer_queue.hasOwnProperty(key)) {
+            if (timer_queue[key].plannedDeleteTime + 10000 > plannedDeleteTime && timer_queue[key].plannedDeleteTime - 10000 < plannedDeleteTime) {
+                needAdd = false
+            } else {
+                clearTimeout(timer_queue[key].timer)
+            }
+        }
+        //If the planned delete time exceeds 12 hours, the thread will not be started
+        if (delayMillisecond > 12 * 1000 * 60 * 60) {
+            needAdd = false
+        }
+        if (needAdd) {
+            //Force delete according to planned deletion time
+            let timer = setTimeout(_deleteSecret, delayMillisecond, DB, appid, secretName)
+            timer_queue[key] = {
+                timer,
+                plannedDeleteTime
+            }
+        }
+    } catch (error) {
+        console.error('error deleteSecret :: ', error)
+    }
+}
+
+/**
+ * Delete secret from the database by secretName
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @returns
+ */
+const _deleteSecret = async (DB, appid, secretName) => {
+    try {
+        const forceDeleteData = require('./secret_manager_apis').forceDeleteData
+        if (await forceDeleteData(DB, appid, secretName)) {
+            console.info(`delete secret thread :: The ${base64_decode(secretName)} delete success`)
+        } else {
+            console.error(`delete secret thread :: The ${base64_decode(secretName)} delete failed`)
+        }
+    } catch (error) {
+        console.error('error _deleteSecret :: ', error)
+    }
+    // remove exist timer
+    removeTimer(secretName, appid, DB)
```

Comment:
does it mean after the secret be deleted, the timer also will be removed？

seems the other existing timer (e,g, nonce_timer, cmk_timer) only will be remove after the service stopped.  can you explain why?